e2etest/zt_blob_index_tags_test.go

```diff
@@ -0,0 +1,176 @@
+	RunScenarios(
+		t,
+		eOperation.Copy(),
+		eTestFromTo.Other(common.EFromTo.LocalBlob()),
+		eValidate.AutoPlusContent(),
+		params{
+			recursive: true,
+			blobTags:  blobTagsStr,
+		},
+		nil,
+		testFiles{
+			defaultSize: "1M",
+			shouldTransfer: []interface{}{
+				//folder("", ),
+				f("file1.txt", with{blobTags: blobTagsStr}),
+			},
+		})
+}
+
+func TestTags_SetTagsSpecialCharactersSingleBlob(t *testing.T) {
+	blobTagsStr := "bla_bla=foo%2b-foo&bla%2fbla%2f2=bar"
+	RunScenarios(
+		t,
+		eOperation.Copy(),
+		eTestFromTo.Other(common.EFromTo.LocalBlob()),
+		eValidate.AutoPlusContent(),
+		params{
+			recursive: true,
+			blobTags:  blobTagsStr,
+		},
+		nil,
+		testFiles{
+			defaultSize: "1M",
+			shouldTransfer: []interface{}{
+				//folder("", ),
+				f("file1.txt", with{blobTags: blobTagsStr}),
+			},
+		})
+}
+
+func TestTags_SetTagsMultipleBlobs(t *testing.T) {
+	blobTagsStr := "foo=bar&blah=bazz"
+	RunScenarios(
+		t,
+		eOperation.Copy(),
+		eTestFromTo.Other(common.EFromTo.LocalBlob()),
+		eValidate.AutoPlusContent(),
+		params{
+			recursive: true,
+			blobTags:  blobTagsStr,
+		},
+		nil,
+		testFiles{
+			defaultSize: "1M",
+			shouldTransfer: []interface{}{
+				folder(""),
+				folder("fdlr1"),
+				f("file1.txt", with{blobTags: blobTagsStr}),
+				f("fdlr1/file2.txt", with{blobTags: blobTagsStr}),
+			},
+		})
+}
+
+func TestTags_PreserveTagsSingleBlob(t *testing.T) {
+	blobTagsStr := "foo/-foo=bar:bar&baz=blah&YeAr=2020"
+	RunScenarios(
+		t,
+		eOperation.Copy(),
+		eTestFromTo.Other(common.EFromTo.BlobBlob()),
+		eValidate.AutoPlusContent(),
+		params{
+			recursive:           true,
+			s2sPreserveBlobTags: true,
+		},
+		nil,
+		testFiles{
+			defaultSize: "1M",
+			shouldTransfer: []interface{}{
+				//folder("", ),
+				f("file1.txt", with{blobTags: blobTagsStr}),
+			},
+		})
+}
+
+func TestTags_PreserveTagsSpecialCharactersSingleBlob(t *testing.T) {
+	RunScenarios(
+		t,
+		eOperation.Copy(),
+		eTestFromTo.Other(common.EFromTo.BlobBlob()),
+		eValidate.AutoPlusContent(),
+		params{
+			recursive:           true,
+			s2sPreserveBlobTags: true,
+		},
+		nil,
+		testFiles{
+			defaultSize: "1M",
+			shouldTransfer: []interface{}{
+				//folder("", ),
+				f("file1.txt", with{blobTags: "foo/-foo=bar:bar&baz=blah&YeAr=2020"}),
+			},
+		})
+}
+
+func TestTags_PreserveTagsMultipleBlobs(t *testing.T) {
+	RunScenarios(
+		t,
+		eOperation.Copy(),
+		eTestFromTo.Other(common.EFromTo.BlobBlob()),
+		eValidate.AutoPlusContent(),
+		params{
+			recursive:           true,
+			s2sPreserveBlobTags: true,
+		},
+		nil,
+		testFiles{
+			defaultSize: "1M",
+			shouldTransfer: []interface{}{
+				folder(""),
+				folder("fdlr1"),
+				f("file1.txt", with{blobTags: "foo=bar&baz=blah&YeAr=2020"}),
+				f("fdlr1/file2.txt", with{blobTags: "temp123=321pmet&zab=halb&rAeY=0202"}),
+			},
+		})
+}
+
+func TestTags_PreserveTagsSpecialCharactersMultipleBlobs(t *testing.T) {
+	RunScenarios(
+		t,
+		eOperation.Copy(),
```

Comment:
>eOperation.Copy(), [](start = 2, length = 18)

enable tests for sync as well

Reply:
We have it `TestTags_PreserveTagsSpecialCharactersDuringSync`  below this one.